### PR TITLE
Update indexes.go

### PR DESCRIPTION
**Pull request by adamcfraser**
Should only set index tombstone flag in xattr mode.

### CODE REVIEW

**Comment by tleyden** [comment]
Original issue:

I'm getting this config / error combination: https://gist.github.com/tleyden/861975c9e58459cd9524a4fb72c7add5

```
2018-04-11T14:24:07.846-07:00 Index+: Attempting to create index using statement: [CREATE INDEX `sg_roleAccess_1` ON `db_no_xattr`(ALL (ARRAY (op.name) FOR op IN OBJECT_PAIRS(`db_no_xattr`._sync.role_access) END)) with {"num_replica":1,"retain_deleted_xattr":true}]
2018-04-11T14:24:07.850-07:00 WARNING: Error creating index sg_roleAccess_1: Error creating index with statement: CREATE INDEX `sg_roleAccess_1` ON `db_no_xattr`(ALL (ARRAY (op.name) FOR op IN OBJECT_PAIRS(`db_no_xattr`._sync.role_access) END)) with {"num_replica":1,"retain_deleted_xattr":true}: [5000] GSI CreateIndex() - cause: Fails to create index.  retain_deleted_xattr can be used only if extended attributes are indexed. - will retry. -- db.(*SGIndex).createIfNeeded.func1() at indexes.go:210
2018-04-11T14:24:07.850-07:00 Debug: RetryLoop retrying Attempt to create index sg_roleAccess_1 after 5 ms.
```

Related to fact that it's not possible to run test num_replicas>0 and retain_deleted_xattr:false in integration/unit tests

Workaround: set num_index_replicas:0 in config

File: db/indexes.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func (i *SGIndex) createIfNeeded(bucket *base.CouchbaseBucketGoCB, useXattrs boo
 	if numReplica > 0 || (useXattrs && i.shouldIndexTombstones()) {
 		options = &base.N1qlIndexOptions{
 			NumReplica:      numReplica,
-			IndexTombstones: i.shouldIndexTombstones(),
+			IndexTombstones: useXattrs && i.shouldIndexTombstones(),
```
Comment (tleyden) [inline db/indexes.go]: Can you add a comment somewhere that explains why indexing tombstones are only needed if `useXattrs` is set to true?

Alternatively, what about changing it to:

```
IndexTombstones: i.shouldIndexTombstones(useXattrs)
```

and having `i.shouldIndexTombstones()` return false if `useXattrs != true`?   I think the advantage there is that any code that needs to make that decision will be forced to think about about whether xattrs are enabled or not.

**Comment by tleyden** [comment]
On [feature/issue_2854_remove_legacy_import_handling](https://github.com/couchbase/sync_gateway/tree/feature/issue_2854_remove_legacy_import_handling) I toggled between having xattrs enabled -> xattrs disabled and I saw the same errors mentioned in https://github.com/couchbase/sync_gateway/pull/3440#issuecomment-380605233 (sorry, not sure if they were exactly the same errors).

I tried to apply the change in this PR, but it didn't seem to fix it.  Maybe I'm on a further edge case than this is supposed to address, but I figured I'd mention it just in case.

I'm seeing these errors:

https://gist.github.com/tleyden/f730da835f41d00cce1ff9d8c4f7f0df#file-sg_logs-txt

and the SG config is:

https://gist.github.com/tleyden/f730da835f41d00cce1ff9d8c4f7f0df#file-sg_config-json

**Comment by adamcfraser** [comment]
@tleyden You can see the error in your latest logs:
"There are not enough indexer nodes to create index with replica count of 1. Some indexer nodes may be marked as excluded."

I'm guessing your server cluster doesn't have the indexing service on more than one node?

File: db/indexes.go (hunk 1 above)
Comment (adamcfraser) [reply db/indexes.go]: There's already a comment included w/ the associated flag:
	IdxFlagIndexTombstones                           // When xattrs=true, index should be created with {“retain_deleted_xattr”:true} in order to index tombstones

I'll add an additional comment w/ shouldIndexTombstones.

**Comment by adamcfraser** [comment]
@tleyden Please have another look.

**Comment by tleyden** [comment]
> I'm guessing your server cluster doesn't have the indexing service on more than one node?

Correct.  There's only a single node in the cluster which is running all of the services.

**Comment by tleyden** [comment]
Notes:

- The config defaults to requiring at least 1 indexer node replica, because that is required for the improved failover handling provided by GSI
- With a single node, there is not enough indexer nodes to create a replica of the index, hence this error
- It can be worked around  by explicitly setting the config to require 0 index replicas, which is not a recommended config

**Review by tleyden — APPROVED** [review]
LGTM

**Pull request merged**